src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -315,14 +315,23 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
         configmap_rg_name = configmap.data["AZURE_RESOURCE_GROUP"]
         configmap_cluster_name = configmap.data["AZURE_RESOURCE_NAME"]
         if connected_cluster_exists(client, configmap_rg_name, configmap_cluster_name):
+            preview_cluster_resource = None
+            public_key = None
```

Comment:
Why are you setting these, if you are anyway overwriting them later, in lines 322 and 324?

Looks like earlier there was some reason to reset these.  Do you know what was it?  Can you confirm that is not broken with your change?

Reply:
On the one hand this is purely for code readability - the values of these variables are set in a try block, where any error happening might cause these variables not to be set to the expected value - this way it would be obvious on first glance what default value is assigned to the variable in such error cases.
Additionally, while I know it is not not necessary for Python, it establishes the "variable scope" outside of the try block, which in my opinion also improves on the readability